sklearn/preprocessing/label.py

```diff
@@ -54,15 +84,45 @@ def _encode_numpy(values, uniques=None, encode=False, check_unknown=True):
         return uniques
 
 
-def _encode_python(values, uniques=None, encode=False):
+class _TableWithNan(object):
```

Comment:
Does this slow down the encoding much?

Reply:
dictionary get 20x slower:
```python
def time_dict(d):
    for x in range(10000):
        d[x] = x+1

%timeit time_dict(dict())
# 793 µs ± 13 µs per loop
%timeit time_dict(_DictWithNan())
# 15.3 ms ± 107 µs per loop
```

but it is 1.4x slower on a vector:
```python
def time_encode():
    x = np.random.choice(range(10),size = 10000)
    _ = _encode_python(x)

%timeit time_encode() # master
# 799 µs ± 2.41 µs per loop 
%timeit time_encode() # this PR
# 1.14 ms ± 18.3 µs per loop
```